drivers/ili9341/ili9341.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2018 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ili9341
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the ili9341 display controller
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include "byteorder.h"
+#include "periph/spi.h"
+#include "xtimer.h"
+#include "ili9341.h"
+#include "ili9341_internal.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static void ili9341_cmd_start(ili9341_t *dev, uint8_t cmd, bool cont)
+{
+    gpio_clear(dev->params.dcx_pin);
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, cont, cmd);
+    gpio_set(dev->params.dcx_pin);
+}
+
+/* datasheet page 178, table converted to equation.
+ * gvdd in 1mv increments: 4850 = 4.85V */
+static uint8_t ili9341_calc_pwrctl1(uint16_t gvdd)
+{
+    return (gvdd - 2850) / 50;
+}
+
+static uint8_t ili9341_calc_vmh(uint16_t vcomh)
+{
+    return (vcomh - 2700) / 25;
+}
+
+static uint8_t ili9341_calc_vml(int16_t vcoml)
+{
+    return (vcoml + 2500) / 25;
+}
+
+void ili9341_write_cmd(ili9341_t *dev, uint8_t cmd, const uint8_t *data,
+                       size_t len)
+{
+    spi_acquire(dev->params.spi, dev->params.cs_pin, SPI_MODE_0,
+                dev->params.spi_clk);
+    ili9341_cmd_start(dev, cmd, len ? true : false);
+    if (len) {
+        spi_transfer_bytes(dev->params.spi, dev->params.cs_pin, false, data,
+                           NULL, len);
+    }
+    spi_release(dev->params.spi);
+}
+
+void ili9341_read_cmd(ili9341_t *dev, uint8_t cmd, uint8_t *data, size_t len)
+{
+    assert(len);
+    spi_acquire(dev->params.spi, dev->params.cs_pin, SPI_MODE_0,
+                dev->params.spi_clk);
+    ili9341_cmd_start(dev, cmd, true);
+    /* Dummy transfer */
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, true, 0x00);
+    spi_transfer_bytes(dev->params.spi, dev->params.cs_pin, false, NULL,
+                       data, len);
+    spi_release(dev->params.spi);
+}
+
+static void ili9341_set_area(ili9341_t *dev, uint16_t x1, uint16_t x2,
```

Comment:
Maybe it is an idea to add this to the public API. Together with a write function that allows one to set `n` bytes in memory this would allow the writing of an area in multiple chunked writes. I have no use case for this, so feel free to ignore.

Reply:
We can always extend the API in a follow up PR :)